### PR TITLE
adjust conditions to enable jit decomposition pass only for GPU device

### PR DESCRIPTION
**Pull request by fengyuan14**
Signed-off-by: Feng Yuan <feng1.yuan@intel.com>

### CODE REVIEW

**Comment by arthuryuan1987** [comment]
[https://github.com/pytorch/pytorch/issues/73639](url)

**Comment by arthuryuan1987** [comment]
#https://github.com/pytorch/pytorch/issues/73639

**Comment by arthuryuan1987** [comment]
The issue is described in #https://github.com/pytorch/pytorch/issues/73639. Thanks.

**Comment by arthuryuan1987** [comment]
The purpose is to enable decomposition pass only for GPU device, here listed two reasons,
1. On our device backend, we hope to have an operator level optimization for BatchNorm and LayerNorm (mkldnn support) on JIT graph. So we don't want to decompose these two operators.
2. According to the current implementation and my understanding, decomposition pass only benefits GPU device, where we can get elementwise kernels fusion by TensorExpr lowering. But current implementation is to enable decomposition for all non-CPU devices,
```
 44   auto device = input->type()->expectRef<TensorType>().device();
 45   // As of now, we do the decomposition for batchnorm/layernorm on GPU device
 46   // only
 47   if (!device || (*device).is_cpu()) {
 48     return false;
 49   }
```

**Comment by arthuryuan1987** [comment]
Hi, @eellison , any comments?

**Review by eellison — APPROVED** [review]
LGTM -

**Comment by arthuryuan1987** [comment]
@eellison Can you help to add a `release notes` and `topics` for the PR?

**Comment by arthuryuan1987** [comment]
Hi, @davidberard98 @eellison Could you please add a `release_notes` and `topics` to the PR? and reopen the PR?

**Comment by davidberard98** [comment]
@arthuryuan1987 added labels. Why did you want to reopen the PR?

**Comment by arthuryuan1987** [comment]
> has imported this pull request

Hi, @davidberard98, I am not familiar with the PR process of PyTorch. Does it mean the commit will be merged by you?

**Comment by davidberard98** [comment]
Ah, the way it works (or at least how it used to work, the process changed a little bit recently) is 1) import PR to internal codebase, 2) merge internally, and 3) a bot rebases the commit onto master.

You can see that the changes got merged here:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/decompose_ops.cpp#L47

**Comment by arthuryuan1987** [comment]
> Ah, the way it works (or at least how it used to work, the process changed a little bit recently) is 1) import PR to internal codebase, 2) merge internally, and 3) a bot rebases the commit onto master.
> 
> You can see that the changes got merged here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/decompose_ops.cpp#L47

Thanks for your patient explanation.
